Update docs/getting_started_make_guide.md (#11373)
qmk/qmk_firmware: docs/getting_started_make_guide.md

The `<target>` means the following
* If no target is given, then it's the same as `all` below
* `all` compiles as many keyboard/revision/keymap combinations as specified. For example, `make planck/rev4:default` will generate a single .hex, while `make planck/rev4:all` will generate a hex for every keymap available to the planck.
* `flash`, `dfu`, `teensy`, `avrdude`, `dfu-util`, or `bootloadHID` compile and upload the firmware to the keyboard. If the compilation fails, then nothing will be uploaded. The programmer to use depends on the keyboard. For most keyboards it's `dfu`, but for ChibiOS keyboards you should use `dfu-util`, and `teensy` for standard Teensys. To find out which command you should use for your keyboard, check the keyboard specific readme. 
  Visit the [Flashing Firmware](flashing.md) guide for more details of the available bootloaders.

>   * **Note**: some operating systems need root access for these commands to work, so in that case you need to run for example `sudo make planck/rev4:default:flash`.

Reviewer: Since it's being touched, would much rather avoid the sudo stuff, and point them at udev stuff.

Author: What should I do?
If you can come up with a draft sentence, I will accept it.

Or will someone fix it with another PullReq?

Author: What should I do?
Do you have any suggestions for corrections?

Reviewer: Maybe: 
```suggestion
  * **Note**: some operating systems need privileged access for these commands to work. This means that you may need to setup [`udev rules`](faq_build.md#linux-udev-rules) to access these without root access, or to run the command with root access (`sudo make planck/rev4:default:flash`).
```

* `clean`, cleans the build output folders to make sure that everything is built from scratch. Run this before normal compilation if you have some unexplainable problems.
* `distclean` removes .hex files and .bin files.

> The following `<target>` is for developers.

Reviewer: ```suggestion
The following `<target>` are for developers.
```

> The following `<target>` are for developers.

Reviewer 2: ```suggestion
The following targets are for developers:
```
